toolchain/parse/handle_decl_name_and_params.cpp

```diff
@@ -27,7 +27,13 @@ static auto HandleDeclNameAndParams(Context& context, State after_name)
     CARBON_DIAGNOSTIC(ExpectedDeclName, Error,
                       "`{0}` introducer should be followed by a name.",
                       Lex::TokenKind);
-    context.emitter().Emit(*context.position(), ExpectedDeclName,
+    Lex::Token location = *context.position();
+    if (context.tokens().GetKind(location) == Lex::TokenKind::EndOfFile) {
```

Comment:
I should note too, there are probably a lot of places that we could get into logic like this. If you want this more generally, a helper function would probably be suitable.

Reply:
Definitely agree that we'll likely want something like this eventually.

That said, I'm not super confident on what it will look like yet... I worry that the code I've used here is somewhat specific to this diagnostic and we might need more interesting logic for others. My preference would be to wait until we have a few cases to look at before trying to generalize if that's OK?

Reply:
Sure, it's not important.